--- FabLabKannai-GoogleAssistant-0.py	2017-11-25 20:30:37.273974467 +0900
+++ FabLabKannai-GoogleAssistant-1.py	2017-11-25 20:43:12.909997909 +0900
@@ -28,6 +28,29 @@
 from google.assistant.library.event import EventType
 from google.assistant.library.file_helpers import existing_file
 
+import subprocess           # コマンドを実行するためのライブラリ
+
+# 音源データ
+SOUND_DIR='/home/pi/sound'
+SOUND_ACK = [
+        SOUND_DIR+'/computerbeep_43.mp3',
+        SOUND_DIR+'/computerbeep_58.mp3',
+        SOUND_DIR+'/computerbeep_12.mp3']
+
+proc = None
+
+# 確認音の再生
+def play_ack(num):
+    global proc
+
+    print('> play_ack(', num, '): ', end='')
+    if proc != None:
+        proc.terminate()
+        print('proc.terminate(): ', end='')
+    cmd = ['cvlc', '-q', '--play-and-exit', SOUND_ACK[num]]
+    proc = subprocess.Popen(cmd)
+    print(cmd)
+
 def process_event(event):
     """Pretty prints events.
 
@@ -42,9 +65,19 @@
 
     print(event)
 
+    if event.type == EventType.ON_START_FINISHED:
+        play_ack(0)
+
+    if event.type == EventType.ON_CONVERSATION_TURN_STARTED:
+        play_ack(1)
+
+    if event.type == EventType.ON_END_OF_UTTERANCE:
+        play_ack(2)
+
     if (event.type == EventType.ON_CONVERSATION_TURN_FINISHED and
             event.args and not event.args['with_follow_on_turn']):
         print()
+        play_ack(0)
 
 
 def main():
