--- FabLabKannai-GoogleAssistant-0.py	2018-02-24 00:20:17.217899386 +0900
+++ FabLabKannai-GoogleAssistant-1.py	2018-02-24 00:44:34.778980744 +0900
@@ -28,9 +28,25 @@
 from google.assistant.library.event import EventType
 from google.assistant.library.file_helpers import existing_file
 
+import subprocess
+
+SOUND_DIR = '/home/pi/sound'
+SOUND_ACK = [
+	SOUND_DIR + '/computerbeep_43.mp3',
+	SOUND_DIR + '/computerbeep_58.mp3',
+	SOUND_DIR + '/computerbeep_12.mp3']
+
+proc = None
 
 DEVICE_API_URL = 'https://embeddedassistant.googleapis.com/v1alpha2'
 
+def play_ack(num):
+	global proc
+
+	if proc != None:
+		proc.terminate()
+	cmd = ['cvlc', '-q', '--play-and-exit', SOUND_ACK[num]]
+	proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
 
 def process_device_actions(event, device_id):
     if 'inputs' in event.args:
@@ -61,9 +77,25 @@
 
     print(event)
 
+    if event.type == EventType.ON_START_FINISHED:
+        play_ack(0)
+
+    if event.type == EventType.ON_CONVERSATION_TURN_STARTED:
+        play_ack(1)
+
+    if event.type == EventType.ON_END_OF_UTTERANCE:
+        play_ack(2)
+
+    if event.type == EventType.ON_CONVERSATION_TURN_TIMEOUT:
+        play_ack(0)
+
+    if event.type == EventType.ON_NO_RESPONSE:
+        play_ack(0)
+
     if (event.type == EventType.ON_CONVERSATION_TURN_FINISHED and
             event.args and not event.args['with_follow_on_turn']):
-        print()
+        play_ack(0)
+
     if event.type == EventType.ON_DEVICE_ACTION:
         for command, params in process_device_actions(event, device_id):
             print('Do command', command, 'with params', str(params))
