--- FabLabKannai-GoogleAssistant-2.py	2018-02-24 00:55:30.219676079 +0900
+++ FabLabKannai-GoogleAssistant-3.py	2018-02-24 00:59:02.059905228 +0900
@@ -40,9 +40,18 @@
 proc = None
 
 PIN_LED = 17
+PIN_BUTTON = 27
+BOUNCE_MSEC = 500
 
 DEVICE_API_URL = 'https://embeddedassistant.googleapis.com/v1alpha2'
 
+def procButton(pin):
+    global assistant
+
+    if pin == PIN_BUTTON:
+            assistant.start_conversation()
+    print()
+
 def play_ack(num):
 	global proc
 
@@ -153,6 +162,9 @@
 
     GPIO.setmode(GPIO.BCM)
     GPIO.setup(PIN_LED, GPIO.OUT)
+    GPIO.setup(PIN_BUTTON, GPIO.IN, GPIO.PUD_DOWN)
+    GPIO.add_event_detect(PIN_BUTTON, GPIO.BOTH,
+       callback=procButton, bouncetime=BOUNCE_MSEC)
 
     parser = argparse.ArgumentParser(
         formatter_class=argparse.RawTextHelpFormatter)
